src/core/Song.cpp

```diff
@@ -947,17 +949,24 @@ void Song::loadProject( const QString & fileName )
 
 	Engine::projectJournal()->setJournalling( false );
 
+	m_oldFileName = m_fileName;
 	m_fileName = fileName;
-	m_oldFileName = fileName;
 
 	DataFile dataFile( m_fileName );
 	// if file could not be opened, head-node is null and we create
 	// new project
-	if( dataFile.validate( fileName.right( fileName.lastIndexOf(".") ) ) )
```

Comment:
If dataFile.validate() is needed here we could keep it in with `an &&` but right now it seem to be kind of broken. The only other place that calls it is in [/src/gui/FileBrowser.cpp](https://github.com/zonkmachine/lmms/blob/9ea7f2eb3419a581651ef71ad11ed7ce366cde04/src/gui/FileBrowser.cpp#L421) but it doesn't seem to execute properly. Try dragging or double clicking on some random non-valid file in the "My Projects" or samples. The message box should appear but it doesn't. It does however if I invert the logic (removes the ! in the linked line above) but that's prone to crashing. For now, here, `dataFile.head().isNull()` should be enough.